### DISCUSSION TITLE
Getting the current loop iteration in Javascript Automation

### DISCUSSION THREAD

**Discussion by Vixiea** [Budibase/budibase#6373] [Help] [answered]
I'm building an automation to query an outside rest data source and then iterate over the length of it to add the item IDs into my local table. While I can get this to work fine for non loops using `return $("steps.1.response")[0].PB_ItemID.toString()` I'm not having any luck using `return $("steps.1.response")[$("loop.currentItem")].PB_ItemID.toString()` to get the current iteration number.

If I only pass  `{{ loop.currentitem }}` I can see it creating items with IDs of the current iteration number, but attempting to run this as `return $("loop.currentitem")` returns an error that it's blank. 

Is there an easier way to iterate through the responses from my rest query or am I missing something? I'm running version 1.0.206.

Edit:
I attempted to use `return "test " + $("loop.currentItem")` and got this output: "ItemID": "test undefined"
Here's my test loop:
![image](https://user-images.githubusercontent.com/70721547/174351560-b037b8d6-fb9d-4d84-8ca8-24308127db33.png)

**Comment by melohagan** [comment, marked as answer]
Hi @Vixiea 

Thanks for reporting. I was able to reproduce the bug, which I have raised here: https://github.com/Budibase/budibase/issues/6387
